src/include/platform/internal/GenericDeviceInstanceInfoProvider.ipp

```diff
@@ -52,6 +52,42 @@ CHIP_ERROR GenericDeviceInstanceInfoProvider<ConfigClass>::GetProductName(char *
     return CHIP_NO_ERROR;
 }
 
+template <class ConfigClass>
```

Comment:
Looks like DeviceInstanceInfoProvider class is half implemented using virtual interface and half using CRTP, this GenericDeviceInstanceInfoProvider.ipp should not exist, I will clean this up in a separate PR

Reply:
I think `GenericDeviceInstanceInfoProvider` doesn't follow CRTP, but is just a common base class for those implementations that still use `ConfigurationManager` under the hood for retrieving this data. Perhaps, it could be removed, but at the cost of more code duplication. Even platforms that provide their factory data providers typically use it as an option, as that requires programming factory data to flash in order to work correctly, so for daily testing it is useful to have some kind of "generic" implementation.
But I agree it would be great if we could get rid of `ConfigurationManager` completely.